File: 01/2.php
<?php

declare(strict_types=1);

const DIGITS_NAME = ['one' => 'o1e', 'two' => 't2o', 'three' => 't3e', 'four' => 'f4r', 'five' => 'f5e', 'six' => 's6x', 'seven' => 's7n', 'eight' => 'e8t', 'nine' => 'n9e'];

$numbers = array_map(function (string $string): int {
    foreach (DIGITS_NAME as $key => $value) {
        $string = str_replace($key, $value, $string);
    }
    $number = preg_replace("/[^0-9]/", "", $string);
    $result = (int) ($number[0].$number[strlen($number)-1]);

    return $result;

}, file('test.txt'));

echo array_sum($numbers).PHP_EOL;

File: 01/2.py
import re

DIGITS_REPLACEMTNS = {'one': 'o1e','one': 'o1e', 'two': 't2o', 'three' : 't3e', 'four' : 'f4r', 'five' : 'f5e', 'six' : 's6x', 'seven' : 's7n', 'eight' : 'e8t', 'nine' : 'n9e'
}

sum = 0
for row in open("test.txt", "r+").read().splitlines():
    for numberName in DIGITS_REPLACEMTNS:
        row = row.replace(numberName, DIGITS_REPLACEMTNS[numberName])
    number = re.sub('[^0-9]', '', row)
    number = number[0]+number[len(number)-1]
    sum += int(number)
    
